src/wechaty.ts

```diff
@@ -114,13 +114,8 @@ export interface WechatyOptions {
   ioToken?       : string,                    // Io TOKEN
 }
 
-// A plugin is a Class with a install function.
-export abstract class WechatyPlugin {
-
-  abstract install (instance: Wechaty): void
-
-  abstract installed?: boolean
-
+export interface WechatyPlugin {
+  (this: Wechaty): void;
```

Comment:
How about let's use an explicit args for passing the `wechaty` instead of using `this`?

I'll write the reason in detail in the comments outside.